### PR TITLE
feat: relaunch browser when switching testing types from app

### PR DESCRIPTION
**Pull request by ZachJW34**
- Closes [UNIFY-923](https://cypress-io.atlassian.net/browse/UNIFY-923)

### User facing changelog
Relaunch browser when switching testing type

### Additional details
Allows the browser to relaunch when switching testing type and also fixes some minor UI issues with the OpenBrowserList (icons).

There was a bug with `reconfigureProject` that could be seen in the terminal when switching from e2e -> component. `reconfigureProject` was calling `initializeActiveProject` but the `ProjectLifeCycleManager` was calling this as well when you switch testing type. The lifecycle manager should be in charge of spawning a new project since it relies on the config being configured properly so I removed the call in `reconfigureProject`.

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/151638320-72b8fd1a-6a29-4ba5-8c79-687a0895b0df.mov

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/launchpad/cypress/e2e/choose-a-browser.cy.ts [hunk 1]
```diff
@@ -117,7 +117,8 @@ describe('Choose a Browser Page', () => {
       }).as('launchProject')
 
       cy.get('@launchButton').click()
-      cy.contains('button', 'Opening E2E Testing in Chrome')
+      // TODO: Relies on ctx.emitter.toLaunchpad which doesn't refresh AUT
```
Comment (ryanthemanuel) [inline packages/launchpad/cypress/e2e/choose-a-browser.cy.ts:120]: Is this work being tracked somewhere?

Comment (ZachJW34) [reply packages/launchpad/cypress/e2e/choose-a-browser.cy.ts:120]: No, I'll create an issue and update this comment. I talked with @tgriesser a bit about this and tried to find out why this wasn't working but had no luck

File: packages/frontend-shared/src/assets/icons/testing-type-component_x16.svg [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<svg data-cy="icon-testing-type-component" width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+  <path d="M2 10H6V14H2V10Z" fill="white" fill-opacity="0.5"/>
+  <path d="M10 2H14V6H10V2Z" fill="white" fill-opacity="0.5"/>
+  <path d="M2 6V5C1.44772 5 1 5.44772 1 6H2ZM6 6H7V5H6V6ZM2 14H1C1 14.5523 1.44772 15 2 15V14ZM10 2V1H9V2H10ZM14 2H15C15 1.44772 14.5523 1 14 1V2ZM14 6V7H15V6H14ZM10 6H9V7H10V6ZM10 10H11V9H10V10ZM10 14V15C10.5523 15 11 14.5523 11 14H10ZM2 7H6V5H2V7ZM5 6V10H7V6H5ZM6 9H2V11H6V9ZM3 10V6H1V10H3ZM5 10V14H7V10H5ZM6 13H2V15H6V13ZM3 14V10H1V14H3ZM10 3H14V1H10V3ZM13 2V6H15V2H13ZM14 5H10V7H14V5ZM11 6V2H9V6H11ZM6 11H10V9H6V11ZM9 10V14H11V10H9ZM10 13H6V15H10V13Z" fill="white"/>
+</svg>
```
Comment (lmiller1990) [inline packages/frontend-shared/src/assets/icons/testing-type-component_x16.svg:5]: Random q: where do the icons/markdown come from?

**Review by lmiller1990 — APPROVED** [review]

File: packages/frontend-shared/src/assets/icons/testing-type-component_x16.svg (hunk 2 above)
Comment (ZachJW34) [reply packages/frontend-shared/src/assets/icons/testing-type-component_x16.svg:5]: From Figma. You can click on the icon and export it as an svg. There is also the design-system page that has all the icons as well

https://www.figma.com/file/1WJ3GVQyMV5e7xVxPg3yID/Design-System?node-id=2448%3A3902

Comment (lmiller1990) [reply packages/frontend-shared/src/assets/icons/testing-type-component_x16.svg:5]: Right, ok! I thought it was coming from some library or plugin, for some reason. This makes sense now.

**Comment by ZachJW34** [comment]
Working on fixing the `cypress-in-cypress` tests. They are passing locally and just passed when rerunning job with `ssh`, but refuse to pass when running normally 😕

**Comment by ZachJW34** [comment]
Figured out why `cypress-in-cypress` tests were failing. The changes I made to `sidebar_navigation.cy.ts` caused side effects that resulted in those tests failing. Guess you shouldn't stub `ctx.actions.project.launchProject` as the function will remain stubbed. The reason it passed when I was sshing into the job was that the order that the specs run was preserved whereas when the run in CI normally, they are parallelized and the order is shuffled, causing `sidebar_navigation.cy.ts` to run before the cypress in cypress tests and leaking the stubbed function into it.

**Review by estrada9166 — APPROVED** [review]

**Pull request merged**